src/MainWindow.cpp

```diff
@@ -0,0 +1,936 @@
+    {
+        // Retrieve the current bill being checked from the map
+        Bill currentBill = *billMapIterator;
+
+        // Set the group label as the bill name
+        m_settings.beginGroup(removeSpaces(currentBill.getName()));
+
+        // Write the bill amount due, due date, and status key-value pairs to the config file
+        m_settings.setValue(m_BILL_AMOUNT_DUE_KEY, currentBill.getAmountDue());
+        m_settings.setValue(m_BILL_DUE_DATE_KEY, currentBill.getDueDate().toString(m_DATE_STRING_FORMAT));
+        m_settings.setValue(m_BILL_FUNDING_STATUS_KEY, fundingStatusBooleanToString(currentBill.isFunded()));
+        m_settings.endGroup();
+    }
+
+    // Sync the settings object at the end
+    m_settings.sync();
+}
+
+void MainWindow::fundBillOnClick()
+{
+    // Iterate over the bill table widget rows
+    for(int row = 0; row < m_billTableWidget->rowCount(); row++)
+    {
+        // Iterate over the columns in each row
+        for(int col = 0; col < m_billTableWidget->columnCount(); col++)
+        {
+            // Store the column header
+            QString columnHeader = m_billTableWidget->horizontalHeaderItem(col)->text();
+
+            // If the current column is the bill's funding status and the row is selected
+            if(columnHeader == m_BILL_FUNDING_STATUS_COLUMN_HEADER_STRING && m_billTableWidget->item(row, 0)->isSelected())
+            {
+                // Retrieve the funding status combo box
+                QComboBox *fundedStatusBox;
+                fundedStatusBox = (QComboBox*)m_billTableWidget->cellWidget(row, col);
+
+                // Switch the funding status to funded
+                fundedStatusBox->setCurrentIndex(1);
+            }
+        }
+    }
+}
+
+void MainWindow::defundBillOnClick()
+{
+    // Iterate over the bill table widget rows
+    for(int row = 0; row < m_billTableWidget->rowCount(); row++)
+    {
+        // Iterate over the columns in each row
+        for(int col = 0; col < m_billTableWidget->columnCount(); col++)
+        {
+            // Store the column header
+            QString columnHeader = m_billTableWidget->horizontalHeaderItem(col)->text();
+
+            // If the current column is the bill's funding status and the row is selected
+            if(columnHeader == m_BILL_FUNDING_STATUS_COLUMN_HEADER_STRING && m_billTableWidget->item(row, 0)->isSelected())
+            {
+                // Retrieve the funding status combo box
+                QComboBox *fundedStatusBox;
+                fundedStatusBox = (QComboBox*)m_billTableWidget->cellWidget(row, col);
+
+                // Switch the funding status to not funded
+                fundedStatusBox->setCurrentIndex(0);
+            }
+        }
+    }
+}
+
+void MainWindow::deleteBillOnClick()
+{
+    // Iterate over the bill table widget rows
+    for(int row = 0; row < m_billTableWidget->rowCount(); row++)
```

Comment:
The nested for loops, storing of the column header, and the initial if statement retrieving the funded status combo box are all repeated across this function as well as `fundBillOnClick` and `defundBillOnClick`. Store this repeated code in a subroutine.

Reply:
Small subroutine added for checking row selection and updating funding status.